File: lazarus/scripts/systems/position_system.js
import { System } from './systems.js'
import { PositionComponent } from '../components/position_component.js'

class PositionSystem extends System {
    get_position(entity, x, y, rot=0, size=1){
	var component = new PositionComponent(entity, x, y, rot, size)
	this.components.push(component)
	return component
    }
    update(){
	this.garbage_collect()
    }
}

export var position_system = new PositionSystem()

File: fiddling/server/systems/position_system.py
from .systems import System
from server.components.position_component import PositionComponent


class PositionSystem(System):
    def get_position(self, entity, x, y, rot=0, size_x=1, size_y=None):
        position = PositionComponent(entity, x, y, rot, size_x, size_y)
        self.components.append(position)
        return position

    def update(self, world, dt):
        self.garbage_collect()


position_system = PositionSystem()
